src/SALib/analyze/delta.py

```diff
@@ -66,15 +75,19 @@ def analyze(
         np.random.seed(seed)
 
     D = problem["num_vars"]
-    N = Y.size
+    if y_resamples is None:
+        y_resamples = Y.size
+    
+    if not y_resamples <= Y.size:
+        raise ValueError("y_resamples must be less-equal than the total  number of samples")

```

Comment:
Remove extra space in error message ("total  number")

"less-equal" -> "less than or equal to"